simulation/src/test/java/com/palantir/dialogue/core/Strategy.java

```diff
@@ -54,114 +48,69 @@ public Channel getChannel(Simulation simulation, Supplier<List<SimulationServer>
         });
     }
 
     private static Channel pinUntilError(Simulation sim, Supplier<List<SimulationServer>> channelSupplier) {
-        Random psuedoRandom = new Random(3218974678L);
+        Random psuedo = new Random(3218974678L);
         return RefreshingChannelFactory.RefreshingChannel.create(channelSupplier, channels -> {
-            List<LimitedChannel> limitedChannels = channels.stream()
-                    .map(addConcurrencyLimiter(sim))
-                    .map(addFixedLimiter(sim))
-                    .collect(Collectors.toList());
-            DialoguePinuntilerrorMetrics metrics = DialoguePinuntilerrorMetrics.of(sim.taggedMetrics());
-            LimitedChannel limited = new PinUntilErrorChannel(
-                    new PinUntilErrorChannel.ReshufflingNodeList(limitedChannels, psuedoRandom, sim.clock(), metrics),
-                    metrics);
-            return retryingChannel(sim, limited);
+            return DialogueChannel.builder()
+                    .channels(channels)
+                    .clientConfiguration(ClientConfiguration.builder()
+                            .from(stubConfig())
+                            .taggedMetricRegistry(sim.taggedMetrics())
+                            .nodeSelectionStrategy(NodeSelectionStrategy.PIN_UNTIL_ERROR)
+                            .build())
+                    .clock(sim.clock())
+                    .random(psuedo)
+                    .scheduler(sim.scheduler())
+                    .build();
         });
     }
 
-    private static Channel roundRobin(Simulation sim, Supplier<List<SimulationServer>> channelSupplier) {
+    private static Channel unlimitedRoundRobin(Simulation sim, Supplier<List<SimulationServer>> channelSupplier) {
+        Random random = new Random(3218974678L);
         return RefreshingChannelFactory.RefreshingChannel.create(channelSupplier, channels -> {
-            List<LimitedChannel> limitedChannels = channels.stream()
-                    .map(Strategy::noOpLimitedChannel)
-                    .map(addFixedLimiter(sim))
-                    .collect(Collectors.toList());
-            LimitedChannel limited = new RoundRobinChannel(limitedChannels);
-            return retryingChannel(sim, limited);
+            return DialogueChannel.builder()
+                    .channels(channels)
+                    .clientConfiguration(ClientConfiguration.builder()
+                            .from(stubConfig())
+                            .taggedMetricRegistry(sim.taggedMetrics())
+                            .nodeSelectionStrategy(NodeSelectionStrategy.ROUND_ROBIN)
+                            .failedUrlCooldown(Duration.ofMillis(200))
+                            .clientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS)
+                            .build())
+                    .clock(sim.clock())
+                    .random(random)
+                    .scheduler(sim.scheduler())
+                    .build();
```

Comment:
I think using the same ClientConfiguration in our simulation makes these a bit more believable.

Reply:
This is fantastic :-)

Reply:
Ok I've factored out all the builder stuff into a (hopefully) non-controversial PR: https://github.com/palantir/dialogue/pull/452